几个Bug

_半期考试忙，好久没发issue了_
_最近感觉issue发多了修起来很累_
1.游戏在显示界面前崩溃后，启动器会卡在“等待游戏启动”
![2017-05-20 11-09-26](https://cloud.githubusercontent.com/assets/17371317/26272522/2df6a694-3d4d-11e7-8e45-fbae1975cb53.png)
游戏日志
```
MC: Exception in thread "main" java.lang.NoClassDefFoundError: com/mojang/authlib/AuthenticationService
MC: 	at net.minecraft.client.main.Main.main(SourceFile:39)
MC: Caused by: java.lang.ClassNotFoundException: com.mojang.authlib.AuthenticationService
MC: 	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
MC: 	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
MC: 	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:335)
MC: 	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
MC: 	... 1 more
```
（库没下载完，很显然是下一个Bug引发的）
2.库没下载完成依然启动了游戏，没有让用户选择
除了forge有些库不再依赖了其他的库必须要下载才能启动
完整日志
```
[11:04:16] [main/INFO] *** Hello Minecraft! Launcher 2.7.6.26 ***
[11:04:17] [main/INFO] Initialized settings.
[11:08:47] [AWT-EventQueue-0/INFO] Start generating launching command...
[11:08:47] [Game Launcher/INFO] Building process
[11:08:47] [Game Launcher/INFO] Logging in...
[11:08:47] [Game Launcher/INFO] Detecting libraries...
[11:08:47] [Task List/INFO] Executing task: PARALLEL
[11:08:47] [pool-2-thread-1/INFO] Executing task: 下载: com.mojang:authlib:1.5.22
[11:08:47] [pool-2-thread-2/INFO] Executing task: 下载: com.mojang:realms:1.9.8
[11:08:47] [pool-2-thread-2/INFO] Downloading: https://libraries.minecraft.net/com/mojang/realms/1.9.8/realms-1.9.8.jar to: /home/su226/.minecraft/libraries/com/mojang/realms/1.9.8/realms-1.9.8.jar
[11:08:47] [pool-2-thread-1/INFO] Downloading: https://libraries.minecraft.net/com/mojang/authlib/1.5.22/authlib-1.5.22.jar to: /home/su226/.minecraft/libraries/com/mojang/authlib/1.5.22/authlib-1.5.22.jar
[11:08:47] [pool-2-thread-1/INFO] Downloading: https://libraries.minecraft.net/com/mojang/authlib/1.5.22/authlib-1.5.22.jar to: /home/su226/.minecraft/libraries/com/mojang/authlib/1.5.22/authlib-1.5.22.jar
[11:08:47] [pool-2-thread-1/INFO] Downloading: https://libraries.minecraft.net/com/mojang/authlib/1.5.22/authlib-1.5.22.jar to: /home/su226/.minecraft/libraries/com/mojang/authlib/1.5.22/authlib-1.5.22.jar
[11:08:47] [pool-2-thread-1/INFO] Downloading: https://libraries.minecraft.net/com/mojang/authlib/1.5.22/authlib-1.5.22.jar to: /home/su226/.minecraft/libraries/com/mojang/authlib/1.5.22/authlib-1.5.22.jar
[11:08:47] [pool-2-thread-1/INFO] Downloading: https://libraries.minecraft.net/com/mojang/authlib/1.5.22/authlib-1.5.22.jar to: /home/su226/.minecraft/libraries/com/mojang/authlib/1.5.22/authlib-1.5.22.jar
[11:08:47] [pool-2-thread-1/INFO] Downloading: https://libraries.minecraft.net/com/mojang/authlib/1.5.22/authlib-1.5.22.jar to: /home/su226/.minecraft/libraries/com/mojang/authlib/1.5.22/authlib-1.5.22.jar
[11:08:47] [pool-2-thread-1/ERROR] Task failed: 下载: com.mojang:authlib:1.5.22
java.io.IOException: 下载失败 https://libraries.minecraft.net/com/mojang/authlib/1.5.22/authlib-1.5.22.jar
	at org.jackhuang.hmcl.util.net.FileDownloadTask.executeTask(FileDownloadTask.java:216)
	at org.jackhuang.hmcl.util.task.TaskList.executeTask(TaskList.java:133)
	at org.jackhuang.hmcl.util.task.TaskList.access$000(TaskList.java:40)
	at org.jackhuang.hmcl.util.task.TaskList$Invoker.run(TaskList.java:80)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.net.UnknownHostException: libraries.minecraft.net
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:184)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:668)
	at sun.net.NetworkClient.doConnect(NetworkClient.java:175)
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:463)
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:558)
	at sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:264)
	at sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:367)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(AbstractDelegateHttpsURLConnection.java:191)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1138)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1032)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:177)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.connect(HttpsURLConnectionImpl.java:153)
	at org.jackhuang.hmcl.util.net.FileDownloadTask.executeTask(FileDownloadTask.java:130)
	... 8 more
[11:08:47] [pool-2-thread-2/INFO] Downloading: https://libraries.minecraft.net/com/mojang/realms/1.9.8/realms-1.9.8.jar to: /home/su226/.minecraft/libraries/com/mojang/realms/1.9.8/realms-1.9.8.jar
[11:08:47] [pool-2-thread-2/INFO] Downloading: https://libraries.minecraft.net/com/mojang/realms/1.9.8/realms-1.9.8.jar to: /home/su226/.minecraft/libraries/com/mojang/realms/1.9.8/realms-1.9.8.jar
[11:08:47] [pool-2-thread-2/INFO] Downloading: https://libraries.minecraft.net/com/mojang/realms/1.9.8/realms-1.9.8.jar to: /home/su226/.minecraft/libraries/com/mojang/realms/1.9.8/realms-1.9.8.jar
[11:08:47] [pool-2-thread-2/INFO] Downloading: https://libraries.minecraft.net/com/mojang/realms/1.9.8/realms-1.9.8.jar to: /home/su226/.minecraft/libraries/com/mojang/realms/1.9.8/realms-1.9.8.jar
[11:08:47] [pool-2-thread-2/INFO] Downloading: https://libraries.minecraft.net/com/mojang/realms/1.9.8/realms-1.9.8.jar to: /home/su226/.minecraft/libraries/com/mojang/realms/1.9.8/realms-1.9.8.jar
[11:08:47] [pool-2-thread-2/ERROR] Task failed: 下载: com.mojang:realms:1.9.8
java.io.IOException: 下载失败 https://libraries.minecraft.net/com/mojang/realms/1.9.8/realms-1.9.8.jar
	at org.jackhuang.hmcl.util.net.FileDownloadTask.executeTask(FileDownloadTask.java:216)
	at org.jackhuang.hmcl.util.task.TaskList.executeTask(TaskList.java:133)
	at org.jackhuang.hmcl.util.task.TaskList.access$000(TaskList.java:40)
	at org.jackhuang.hmcl.util.task.TaskList$Invoker.run(TaskList.java:80)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.net.UnknownHostException: libraries.minecraft.net
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:184)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:668)
	at sun.net.NetworkClient.doConnect(NetworkClient.java:175)
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:463)
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:558)
	at sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:264)
	at sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:367)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(AbstractDelegateHttpsURLConnection.java:191)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1138)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1032)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:177)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.connect(HttpsURLConnectionImpl.java:153)
	at org.jackhuang.hmcl.util.net.FileDownloadTask.executeTask(FileDownloadTask.java:130)
	... 8 more
[11:08:47] [Task List/INFO] Task finished: PARALLEL
[11:08:47] [Task List/INFO] Tasks are successfully finished.
[11:08:47] [Game Launcher/INFO] Unpacking natives...
[11:08:47] [Game Launcher/INFO] *** Make shell command ***
[11:08:47] [Game Launcher/INFO] On making head command.
[11:08:48] [Game Launcher/INFO] Java Version: 1.8.0_131
[11:08:48] [Game Launcher/INFO] Java Platform: 64
[11:08:48] [Game Launcher/INFO] System Platform: 64
[11:08:48] [Game Launcher/INFO] System Physical Memory: 3848
[11:08:49] [Game Launcher/WARN] No assets index file /home/su226/.minecraft/assets/virtual/1.10; can't reconstruct assets
[11:08:49] [Game Launcher/INFO] On making launcher args.
[11:08:49] [Game Launcher/INFO] Starting process
[11:08:49] [Game Launcher/INFO] [/usr/lib/jvm/java-8-oracle/jre/bin/java, -XX:+UseG1GC, -XX:-UseAdaptiveSizePolicy, -XX:-OmitStackTraceInFastThrow, -Xmn128m, -Xmx512m, -Djava.library.path=/home/su226/.minecraft/versions/1.10.2/1.10.2-natives, -Dfml.ignoreInvalidMinecraftCertificates=true, -Dfml.ignorePatchDiscrepancies=true, -Duser.home=/home/su226, -cp, /home/su226/.minecraft/libraries/com/mojang/netty/1.6/netty-1.6.jar:/home/su226/.minecraft/libraries/oshi-project/oshi-core/1.1/oshi-core-1.1.jar:/home/su226/.minecraft/libraries/net/java/dev/jna/jna/3.4.0/jna-3.4.0.jar:/home/su226/.minecraft/libraries/net/java/dev/jna/platform/3.4.0/platform-3.4.0.jar:/home/su226/.minecraft/libraries/com/ibm/icu/icu4j-core-mojang/51.2/icu4j-core-mojang-51.2.jar:/home/su226/.minecraft/libraries/net/sf/jopt-simple/jopt-simple/4.6/jopt-simple-4.6.jar:/home/su226/.minecraft/libraries/com/paulscode/codecjorbis/20101023/codecjorbis-20101023.jar:/home/su226/.minecraft/libraries/com/paulscode/codecwav/20101023/codecwav-20101023.jar:/home/su226/.minecraft/libraries/com/paulscode/libraryjavasound/20101123/libraryjavasound-20101123.jar:/home/su226/.minecraft/libraries/com/paulscode/librarylwjglopenal/20100824/librarylwjglopenal-20100824.jar:/home/su226/.minecraft/libraries/com/paulscode/soundsystem/20120107/soundsystem-20120107.jar:/home/su226/.minecraft/libraries/io/netty/netty-all/4.0.23.Final/netty-all-4.0.23.Final.jar:/home/su226/.minecraft/libraries/com/google/guava/guava/17.0/guava-17.0.jar:/home/su226/.minecraft/libraries/org/apache/commons/commons-lang3/3.3.2/commons-lang3-3.3.2.jar:/home/su226/.minecraft/libraries/commons-io/commons-io/2.4/commons-io-2.4.jar:/home/su226/.minecraft/libraries/commons-codec/commons-codec/1.9/commons-codec-1.9.jar:/home/su226/.minecraft/libraries/net/java/jinput/jinput/2.0.5/jinput-2.0.5.jar:/home/su226/.minecraft/libraries/net/java/jutils/jutils/1.0.0/jutils-1.0.0.jar:/home/su226/.minecraft/libraries/com/google/code/gson/gson/2.2.4/gson-2.2.4.jar:/home/su226/.minecraft/libraries/com/mojang/authlib/1.5.22/authlib-1.5.22.jar:/home/su226/.minecraft/libraries/com/mojang/realms/1.9.8/realms-1.9.8.jar:/home/su226/.minecraft/libraries/org/apache/commons/commons-compress/1.8.1/commons-compress-1.8.1.jar:/home/su226/.minecraft/libraries/org/apache/httpcomponents/httpclient/4.3.3/httpclient-4.3.3.jar:/home/su226/.minecraft/libraries/commons-logging/commons-logging/1.1.3/commons-logging-1.1.3.jar:/home/su226/.minecraft/libraries/org/apache/httpcomponents/httpcore/4.3.2/httpcore-4.3.2.jar:/home/su226/.minecraft/libraries/it/unimi/dsi/fastutil/7.0.12_mojang/fastutil-7.0.12_mojang.jar:/home/su226/.minecraft/libraries/org/apache/logging/log4j/log4j-api/2.0-beta9/log4j-api-2.0-beta9.jar:/home/su226/.minecraft/libraries/org/apache/logging/log4j/log4j-core/2.0-beta9/log4j-core-2.0-beta9.jar:/home/su226/.minecraft/libraries/org/lwjgl/lwjgl/lwjgl/2.9.4-nightly-20150209/lwjgl-2.9.4-nightly-20150209.jar:/home/su226/.minecraft/libraries/org/lwjgl/lwjgl/lwjgl_util/2.9.4-nightly-20150209/lwjgl_util-2.9.4-nightly-20150209.jar:/home/su226/.minecraft/versions/1.10.2/1.10.2.jar, net.minecraft.client.main.Main, --username, su226, --version, HMCL 2.7.6.26, --gameDir, /home/su226/.minecraft, --assetsDir, /home/su226/.minecraft/assets, --assetIndex, 1.10, --uuid, c5b6bd6956cc0f2ef4d28cb3bcce6394, --accessToken, c5b6bd6956cc0f2ef4d28cb3bcce6394, --userType, Legacy, --versionType, HMCL 2.7.6.26, --height, 480, --width, 854]
[11:08:49] [Game Launcher/INFO] Have started the process
MC: Exception in thread "main" java.lang.NoClassDefFoundError: com/mojang/authlib/AuthenticationService
MC: 	at net.minecraft.client.main.Main.main(SourceFile:39)
MC: Caused by: java.lang.ClassNotFoundException: com.mojang.authlib.AuthenticationService
MC: 	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
MC: 	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
MC: 	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:335)
MC: 	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
MC: 	... 1 more
[11:08:50] [ProcessMonitor/INFO] Process exit code: 1
[11:08:50] [ProcessMonitor/ERROR] The game exited abnormally, exit code: 1
[11:08:50] [ProcessMonitor/INFO] Process exit code: 1
[11:08:50] [ProcessMonitor/ERROR] 游戏非正常退出，请查看日志文件，或联系他人寻求帮助。 exit code: 1
[11:08:50] [ProcessMonitor/INFO] Launcher will not exit now.
```
最终导致了崩溃，弹了一堆框框
3.使用Default配置时库文件和资源文件位置错误
依然下载到了`/home/用户名/.minecraft`里面
Windows也一样，下载到了`%appdata%/.minecraft`里
_我在学校机房偷偷下MC，走了才发现库文件和资源文件“不见了”_
4.`(强迫症)`导入和导出整合包的按钮在禁止后样子变了
![2017-05-20 11-09-26](https://cloud.githubusercontent.com/assets/17371317/26272522/2df6a694-3d4d-11e7-8e45-fbae1975cb53.png)
仔细看
5.`(强迫症)`禁用动画后主页依然有动画
学校那软件渲染的虚拟机死慢
6.`(强迫症)`Gnome应用菜单显示问题
显示的是`org-jackhuang-hmcl-Main`
![2017-05-20 11-09-26](https://cloud.githubusercontent.com/assets/17371317/26272522/2df6a694-3d4d-11e7-8e45-fbae1975cb53.png)
再看看MC
![2017-05-20 11-25-54](https://cloud.githubusercontent.com/assets/17371317/26272602/33b5cd4c-3d4f-11e7-90e1-874ba93ebdf9.png)
显示的是`Minecraft 版本号`

--- a/HMCL/src/main/java/org/jackhuang/hmcl/setting/Config.java
+++ b/HMCL/src/main/java/org/jackhuang/hmcl/setting/Config.java
@@ -43,7 +43,6 @@ import org.jackhuang.hmcl.util.javafx.PropertyUtils;
 import org.jackhuang.hmcl.util.platform.OperatingSystem;
 import org.jetbrains.annotations.Nullable;
 
-
 import java.io.File;
 import java.net.Proxy;
 import java.util.Map;

--- a/HMCL/src/main/java/org/jackhuang/hmcl/setting/Config.java
+++ b/HMCL/src/main/java/org/jackhuang/hmcl/setting/Config.java
@@ -146,7 +146,7 @@ public final class Config implements Cloneable, Observable {
     private ObservableList<Map<Object, Object>> accountStorages = FXCollections.observableArrayList();
 
     @SerializedName("fontFamily")
-    private StringProperty fontFamily = new SimpleStringProperty("Consolas");
+    private StringProperty fontFamily = new SimpleStringProperty("Monospace");
 
     @SerializedName("fontSize")
     private DoubleProperty fontSize = new SimpleDoubleProperty(12);
@@ -188,6 +188,10 @@ public final class Config implements Cloneable, Observable {
     private transient ObservableHelper helper = new ObservableHelper(this);
 
     public Config() {
+        // Default override for better-looking logs on Windows
+        if (System.getProperty("os.name").startsWith("Windows")) {
+            this.fontFamily = new SimpleStringProperty("Consolas");
+        }
         PropertyUtils.attachListener(this, helper);
     }
 

--- a/HMCL/src/main/java/org/jackhuang/hmcl/setting/Config.java
+++ b/HMCL/src/main/java/org/jackhuang/hmcl/setting/Config.java
@@ -190,7 +190,7 @@ public final class Config implements Cloneable, Observable {
     public Config() {
         // Default override for better-looking logs on Windows
         if (System.getProperty("os.name").startsWith("Windows")) {
-            this.fontFamily = new SimpleStringProperty("Consolas");
+            this.fontFamily.setValue("Consolas");
         }
         PropertyUtils.attachListener(this, helper);
     }

--- a/HMCL/src/main/java/org/jackhuang/hmcl/setting/Config.java
+++ b/HMCL/src/main/java/org/jackhuang/hmcl/setting/Config.java
@@ -40,7 +40,9 @@ import org.jackhuang.hmcl.util.i18n.Locales;
 import org.jackhuang.hmcl.util.i18n.Locales.SupportedLocale;
 import org.jackhuang.hmcl.util.javafx.ObservableHelper;
 import org.jackhuang.hmcl.util.javafx.PropertyUtils;
+import org.jackhuang.hmcl.util.platform.OperatingSystem;
 import org.jetbrains.annotations.Nullable;
+
 
 import java.io.File;
 import java.net.Proxy;
@@ -146,7 +148,7 @@ public final class Config implements Cloneable, Observable {
     private ObservableList<Map<Object, Object>> accountStorages = FXCollections.observableArrayList();
 
     @SerializedName("fontFamily")
-    private StringProperty fontFamily = new SimpleStringProperty("Monospace");
+    private StringProperty fontFamily = new SimpleStringProperty(OperatingSystem.CURRENT_OS == OperatingSystem.WINDOWS ? "Consolas" : "Monospace");
 
     @SerializedName("fontSize")
     private DoubleProperty fontSize = new SimpleDoubleProperty(12);
@@ -188,10 +190,6 @@ public final class Config implements Cloneable, Observable {
     private transient ObservableHelper helper = new ObservableHelper(this);
 
     public Config() {
-        // Default override for better-looking logs on Windows
-        if (System.getProperty("os.name").startsWith("Windows")) {
-            this.fontFamily.setValue("Consolas");
-        }
         PropertyUtils.attachListener(this, helper);
     }
 

